presto-main/src/main/java/com/facebook/presto/execution/SqlQueryExecution.java

```diff
@@ -160,6 +166,13 @@ public SqlQueryExecution(QueryId queryId,
                 }
             });
 
+            stateMachine.addStateChangeListener(state -> {
```

Comment:
This needs to happen in `start()` so that it's synchronous. State change listeners are async